dictdiffer/__init__.py

```diff
@@ -197,7 +198,8 @@ def check(key):
 
     else:
         # Compare string and integer types and yield `change` flag.
-        if first != second:
+        nan_eq = isinstance(first, float) and math.isnan(first) and isinstance(second, float) and math.isnan(second)
+        if first != second and not nan_eq:
```

Comment:
please see https://github.com/inveniosoftware/dictdiffer/pull/55#issuecomment-145604632

If you have any test case that it is not covered please let me know. Thanks